src/variables.js

```diff
@@ -0,0 +1,266 @@
+import { useState, useEffect } from 'react';
+
```

Comment:
It looks like this one file stores every helper function/array/component you need for the other pages.
I would recommend moving every helper into it's own file, grouping when they make sense (like you could put the post-related components in the same file). And then move all of these files into a subdirectory of `src`.